packages/rollup/src/generators/convert-to-inferred/convert-to-inferred.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  formatFiles,
+  getProjects,
+  type Tree,
+  updateProjectConfiguration,
+} from '@nx/devkit';
+import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';
+
+import { extractRollupConfigFromExecutorOptions } from './lib/extract-rollup-config-from-executor-options';
+import { RollupExecutorOptions } from '../../executors/rollup/schema';
+
+interface Schema {
+  project?: string;
+  all?: boolean;
+  skipFormat?: boolean;
+}
+
+export async function convertToInferred(tree: Tree, options: Schema) {
+  let migrated = 0;
+
+  const projects = getProjects(tree);
+  const originalDefaultOptions = new Map<string, RollupExecutorOptions>();
+
+  forEachExecutorOptions<RollupExecutorOptions>(
+    tree,
+    '@nx/rollup:rollup',
+    (currentTargetOptions, projectName, targetName, configurationName) => {
+      if (!options.all && options.project && projectName !== options.project)
```

Comment:
As above, we can shorten this to `options.project && projectName !== options.project`.

A user specifying `--project` should be the only flag to determine if all projects are migrated, or if a specific project is migrated